src/services/htmlHover.ts

```diff
@@ -62,6 +113,53 @@ export function doHover(document: TextDocument, position: Position, htmlDocument
 	if (tagRange) {
 		return getTagHover(node.tag, tagRange, true);
 	}
+
```

Comment:
This can be done with a single scan start from node.start.

- scan as long as we found a token `scanner.getTokenEnd() >= offset && offset <= scanner.getTokenEnd()`
- if token is tag, attrname or attr value, create the corresponding hover
- while scanning, remember the last seen attribute name